packages/engine-render/src/components/sheets/extensions/row-header-layout.ts

```diff
@@ -71,23 +134,68 @@ export class RowHeaderLayout extends SheetExtension {
+                bottom: rowEndPosition,
+                width: rowHeaderWidth,
+                height: rowEndPosition - preRowPosition,
+            };
+            const [curRowCfg, specStyle] = this.getCfgOfCurrentRow(r);
+
+            // background
+            if (specStyle && curRowCfg.backgroundColor) {
+                ctx.save();
+                ctx.fillStyle = curRowCfg.backgroundColor;
+                ctx.fillRectByPrecision(cellBound.left, cellBound.top, cellBound.width, cellBound.height);
+                ctx.restore();
+            }
+
+            // horizontal line border
+            ctx.beginPath();
+            ctx.moveToByPrecision(cellBound.left, cellBound.bottom);
+            ctx.lineToByPrecision(cellBound.right, cellBound.bottom);
+            ctx.stroke();
+
+            // row header text
+            const textX = (() => {
+                switch (curRowCfg.textAlign) {
+                    case 'center':
+                        return cellBound.left + (cellBound.right - cellBound.left) / 2;
+                    case 'right':
+                        return cellBound.right - MIDDLE_CELL_POS_MAGIC_NUMBER;
+                    case 'left':
+                        return cellBound.left + MIDDLE_CELL_POS_MAGIC_NUMBER;
+                    default: // center
+                        return cellBound.left + (cellBound.right - cellBound.left) / 2;
+                }
+            })();
+            const middleYCellRect = preRowPosition + (rowEndPosition - preRowPosition) / 2 + MIDDLE_CELL_POS_MAGIC_NUMBER; // Magic number 1, because the vertical alignment appears to be off by 1 pixel
+
+            if (specStyle) {
+                ctx.save();
+                ctx.beginPath();
+                this.setStyleToCtx(ctx, curRowCfg);
+                ctx.rectByPrecision(cellBound.left, cellBound.top, cellBound.width, cellBound.height);
+                ctx.clip();
+            }
+
+            ctx.fillText(curRowCfg.text, textX, middleYCellRect);
+            if (specStyle) {
+                ctx.restore();
             }
-            ctx.moveToByPrecision(0, rowEndPosition);
-            ctx.lineToByPrecision(rowHeaderWidth, rowEndPosition);
 
-            const middleCellPos = preRowPosition + (rowEndPosition - preRowPosition) / 2;
-            ctx.fillText(`${r + 1}`, rowHeaderWidth / 2, middleCellPos + MIDDLE_CELL_POS_MAGIC_NUMBER); // Magic number 1, because the vertical alignment appears to be off by 1 pixel.
             preRowPosition = rowEndPosition;
         }
-        // console.log('xx2', rowColumnIndexRange, bounds, this._rowTotalHeight, this._rowHeightAccumulation);
 
-        // painting line bottom border
+        // border right line
         const rowHeaderWidthFix = rowHeaderWidth - 0.5 / scale;
-
+        ctx.beginPath();
         ctx.moveToByPrecision(rowHeaderWidthFix, 0);
-        ctx.lineToByPrecision(rowHeaderWidthFix, rowTotalHeight);
+        ctx.lineToByPrecision(0, rowTotalHeight);
```

Comment:
this should be ctx.lineToByPrecision(rowHeaderWidthFix, rowTotalHeight);

Reply:
Thanks for your review, very useful